pkg/streamingpromql/functions/math.go

```diff
@@ -63,3 +63,23 @@ var UnaryNegation InstantVectorSeriesFunction = func(seriesData types.InstantVec
 
 	return seriesData, nil
 }
+
+var Clamp InstantVectorSeriesFunction = func(seriesData types.InstantVectorSeriesData, scalarArgsData []types.ScalarData, memoryConsumptionTracker *limiting.MemoryConsumptionTracker) (types.InstantVectorSeriesData, error) {
+	outputIdx := 0
+	for step, data := range seriesData.Floats {
+		minVal := scalarArgsData[0].Samples[step].F
+		maxVal := scalarArgsData[1].Samples[step].F
```

Comment:
`scalarArgsData[0]` and `scalarArgsData[1]` could be moved outside the loop, this should save two slice bounds checks on every sample.